python/dronin/telemetry.py

```diff
@@ -629,38 +678,39 @@ def _do_io(self, finish_time):
 
         r,w,e = select.select(rd_set, wr_set, [], tm)
 
-        if r:
-            # Shouldn't throw an exception-- they just told us
-            # it was ready for read.
-            # TODO: Figure out why read sometimes fails when using sockets
-            try:
-                chunk = os.read(self.fd, 1024)
-                if chunk == b'':
-                    raise RuntimeError("stream closed")
-
-                self.recv_buf = self.recv_buf + chunk
+        with self.cond:
```

Comment:
The lock is obtained here so that it is possible to send stuff *either* from the connection thread or outside, so that we don't need to wait to fall out of the select loop to send an object.  The only real alternative is to have our own FD to trigger activity on (loopback pipe) or to do something with signals or...  This is likely to be portable, at least.